packages/nimble-components/src/wafer-map/specs/features/data-manager-refactor.md

```diff
@@ -0,0 +1,73 @@
+# Nimble Wafer Map ataManager Refactor
```

Comment:
The table is addressing a similar issue as the wafermap. The table had no batching of any changes and did updates on every property change while the wafer map batched every property change and recalculated pretty much everything from batch updates.

The table now has an object that tracks changes and triggers the batch execution. During the batch execution the table uses the tracked changes to only update what it needs to. I'd like to see if the wafer map can use a similar pattern: https://github.com/ni/nimble/pull/1099

Reply:
Unresolving to continue the discussion. In the latest update I see the proposed generic UpdateTracker. My suggestion was that we align with the behavior of the table UpdateTracker and I don't think the approach being shown here aligns with the intentions of the table UpdateTracker.

The table update tracker has several major behaviors:
1. Decouple the state change from the needed updates. The decoupling is necessary because one potential state change could lead to multiple unrelated invalidations.
2. Declare how the ordering of updates needs to be defined in one spot
3. Strong typing for the available ordering (states are not tracked by strings, it is very clear what possible updates exist)

Decouple the state change from the needed updates:
Notice how the table flags are not talking about the work needed for the update that happened, but just track that the update happened

https://github.com/ni/nimble/blob/523d60f92bf266d9ecee680762d7ddd606638af8/packages/nimble-components/src/table/models/update-tracker.ts#L36-L43

i.e. it doesn't encode that for rowId changes need to run steps a,b,c in the update tracker, instead it is tracking that hey the rows updated or the columnids updated. Because those things updated they need to be responded to.

Declare how the ordering of updates needs to be defined in one spot:
Now the table has a single update method that can check all the updated states and perform the needed actions.

https://github.com/ni/nimble/blob/523d60f92bf266d9ecee680762d7ddd606638af8/packages/nimble-components/src/table/index.ts#L201-L214

This method is what encodes the order dependence. Not the tracked states themselves. The method now can say if anything related to tanstack updates happened, make sure to update that first, followed by slots, followed by columns.

There is only one spot that defines the ordering for recalculation, not many different spots determining what actions to queue and in what order.